joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -961,6 +966,15 @@ controller_interface::CallbackReturn JointTrajectoryController::on_activate(
     cmd_timeout_ = 0.0;
   }
 
+  {
+    if (get_update_rate() == 0)
+    {
+      throw std::runtime_error("Controller's update rate is set to 0. This should not happen!");
+    }
+    update_period_ =
+      rclcpp::Duration(0.0, static_cast<uint32_t>(1.0e9 / static_cast<double>(get_update_rate())));
+  }
+
```

Comment:
Maybe this can go inside the `on_configure`? What do you think? @fmauch

Reply:
That seems reasonable, yes. I will change that.